### PR TITLE
Fix 1991 admin samples update sample id

### PR DESCRIPTION
**Pull request by alli83**
# Pull request for issue: #1991

This is a pull request for the following functionalities:

* update sample id

## How to test?

Go to https://gigadb.org/adminSample/view/id/187284
Update the Sample ID from [HiC_lib of NAS] to [HiC_lib_of_NAS]
Click 'Save'
The Sample ID is now [HiC_lib_of_NAS]

## How have functionalities been implemented?

Controller updated

## Any changes to automated tests?

adminSampleCest test added

## Any improvements to CI/CD pipeline?

Describe any improvements to the Gitlab pipeline

### CODE REVIEW

**Review by kencho51 — COMMENTED** [review]
Hi @alli83,

**praise:** Can update sample ID in the adminSample form
**question:** Is it better to have a simple validation on the sample id input? Currently, even empty sample ID is allowed be saved
**suggestion:** Regarding the functional test you implemented, I think acceptance will be more suitable, because this fix involves interacting with the admin Sample form, acceptance tests simulate real user interactions with the application and are better suited for ensuring that UI actions, such as updating a form field and clicking "Save," behave as expected

File: tests/acceptance/AdminSample.feature [hunk 1]
```diff
@@ -68,6 +68,14 @@ Feature: admin page for samples
...
     And I should see "Attribute name for the input plant=\rose\ is not valid - please select a valid attribute name!"
 
+  @ok
+  Scenario: display input format error when update
+    Given I am on "/adminSample/update/id/432"
+    When I fill in the field of "name" "Sample[name]" with "test"
+    And I press the button "Save"
+    And I wait "1" seconds
+    Then I should see "test"
+
```
Comment (kencho51) [inline tests/acceptance/AdminSample.feature]: Thanks for adding the acceptance test, but regarding the title of this scenario, I think it is misleading as it looks to me is for testing for the invalid sample ID input.

**suggestion:** Is it better to have 2 scenarios, for example as below:
```
Scenario: display updated sample name when update
    Given I am on "/adminSample/update/id/432"
    When I fill in the field of "name" "Sample[name]" with "test"
    And I press the button "Save"
    And I wait "1" seconds
    Then I should see "test"

Scenario: display error/warning message of empty sample name when update
    Given I am on "/adminSample/update/id/432"
    When I fill in the field of "name" "Sample[name]" with ""
    And I press the button "Save"
    Then I should see "Empty sample name input is not alllowed" (something like this)
  ```

Comment (alli83) [reply tests/acceptance/AdminSample.feature]: my mistake, it was just a copy/paste and I forgot to update the name 😅

**Review by pli888 — COMMENTED** [review]
**praise**: Functionality is working on my dev and staging environments.
**thought**: The functional test `AdminSampleCest` seems to be doing what the acceptance test is or should be doing. I would remove `AdminSampleCest`.
**thought**: I agree with @kencho51's comments regarding acceptance tests.

File: tests/acceptance/AdminSample.feature (hunk 1 above)
Comment (pli888) [reply tests/acceptance/AdminSample.feature]: Ken's first scenario could be improved upon by checking that you are re-directed to the sample view page after successful update of Sample ID using:
```
Then I am on "/adminSample/view/id/432"
And I should see "test"
```

File: tests/acceptance/AdminSample.feature [hunk 2]
```diff
@@ -68,6 +68,24 @@ Feature: admin page for samples
...
+    Given I am on "/adminSample/update/id/432"
+    When I fill in the field of "name" "Sample[name]" with "test"
+    And I press the button "Save"
+    And I wait "1" seconds
+    Then I should see "test"
+    Then I am on "/adminSample/view/id/432"
+    And I should see "test"
+
+  @ok
+  Scenario: display error with an em^ty name
```
Comment (kencho51) [inline tests/acceptance/AdminSample.feature]: **typo:** Should be `empty`

**Review by kencho51 — COMMENTED** [review]
**praise:** The sample ID can be updated now, and a simple input validation has been implemented.
**praise:** The acceptance tests are updated and passing.
**chore:** After typo in the `  Scenario: display error with an em^ty name`, will be happy to approve.

**Review by kencho51 — APPROVED** [review]
LGTM, happy to approve

**Review by pli888 — APPROVED** [review]
**praise**: Acceptance tests are passing.
**praise**: Functionality is working on my dev and staging environments.
**praise**: Code in actionUpdate function in AdminSampleController class is more clear now.
**chore**: Need to rebase with develop branch to remove conflict in CHANGELOG.md.

Happy to approve after rebasing.

**Pull request merged**